program/tests/test_create_mint.rs

```diff
@@ -0,0 +1,773 @@
+fn test_idempotency_false_with_existing_account() {
+    let program_id = spl_token_wrap::id();
+    let mut mollusk = Mollusk::new(&program_id, "spl_token_wrap");
+    mollusk_svm_programs_token::token2022::add_program(&mut mollusk);
+
+    let unwrapped_mint_address = Pubkey::new_unique();
+    let wrapped_token_program_id = spl_token_2022::id();
+    let wrapped_mint_address =
+        get_wrapped_mint_address(&unwrapped_mint_address, &wrapped_token_program_id);
+    let wrapped_backpointer_address = get_wrapped_mint_backpointer_address(&wrapped_mint_address);
+
+    let instruction = create_mint(
+        &program_id,
+        &wrapped_mint_address,
+        &wrapped_backpointer_address,
+        &unwrapped_mint_address,
+        &wrapped_token_program_id,
+        false,
+    );
+
+    // Simulating existing data on mint or backpointer
+    let account_with_data = Account {
+        data: vec![1; 10],
+        ..Account::default()
+    };
+
+    // idempotent: false causes these to fail
+    let accounts = &[
+        (wrapped_mint_address, account_with_data.clone()), // mint already exists
+        (wrapped_backpointer_address, Account::default()),
+        (unwrapped_mint_address, Account::default()),
+        (system_program::id(), Account::default()),
+        mollusk_svm_programs_token::token2022::keyed_account(),
+    ];
+    mollusk.process_and_validate_instruction(
+        &instruction,
+        accounts,
+        &[Check::err(ProgramError::AccountAlreadyInitialized)],
+    );
+
+    let accounts = &[
+        (wrapped_mint_address, Account::default()),
+        (wrapped_backpointer_address, account_with_data), // backpointer already exists
+        (unwrapped_mint_address, Account::default()),
+        (system_program::id(), Account::default()),
+        mollusk_svm_programs_token::token2022::keyed_account(),
+    ];
+    mollusk.process_and_validate_instruction(
+        &instruction,
+        accounts,
+        &[Check::err(ProgramError::AccountAlreadyInitialized)],
+    );
+}
+
+#[test]
+fn test_idempotency_true_with_existing_valid_account() {
+    let program_id = spl_token_wrap::id();
+    let mut mollusk = Mollusk::new(&program_id, "spl_token_wrap");
+    mollusk_svm_programs_token::token2022::add_program(&mut mollusk);
+
+    let unwrapped_mint_address = Pubkey::new_unique();
+    let wrapped_token_program_id = spl_token_2022::id();
+    let wrapped_mint_address =
+        get_wrapped_mint_address(&unwrapped_mint_address, &wrapped_token_program_id);
+    let wrapped_backpointer_address = get_wrapped_mint_backpointer_address(&wrapped_mint_address);
+
+    let instruction = create_mint(
+        &program_id,
+        &wrapped_mint_address,
+        &wrapped_backpointer_address,
+        &unwrapped_mint_address,
+        &wrapped_token_program_id,
+        true,
+    );
+
+    // Simulating existing data on mint or backpointer
+    let mint_account_with_data = Account {
+        data: vec![1; 10],
+        owner: program_id,
```

Comment:
Per the previous comment, this needs to change too
```suggestion
        owner: wrapped_token_program_id,
```